src/ios/FirebaseAnalyticsPlugin.m

```diff
@@ -143,6 +143,27 @@ - (void)showTrackingAuthorizationPopup:(CDVInvokedUrlCommand *)command {
     [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
 }
 
+- (void)setConsent:(CDVInvokedUrlCommand*)command
+{
+    CDVPluginResult* pluginResult = nil;
+    NSError *error;
+    NSDictionary *consentModel = [OSFANLConsentHelper createConsentModel:command.arguments error:&error];
+    if (error) {
+        NSDictionary *errorInfo = error.userInfo;
+        NSString *code = errorInfo[@"code"] ?: @"UNKNOWN";
+        NSString *message = errorInfo[@"message"] ?: error.localizedDescription;
+        
+        NSString *jsonString = [NSString stringWithFormat:@"{\"code\":\"%@\",\"message\":\"%@\"}", code, message];
+        NSLog(@"Error: %@", jsonString);
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR
+                                         messageAsString:jsonString];
```

Comment:
Why are we splitting the Dictionary to then create it again (as is) as a String? Wouldn't all of this be simplified by just using the following `[self sendError:error forCallbackId:command.callbackId];` ?
We could even remove the `pluginResult` local variable and the `[self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];` call if we change the `else` statement as well.